src/app/Services/ShareService.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Services;
+
+use App\Http\Requests\ShareRequest;
+use App\Models\Post;
+
+class ShareService
+{
+    /**
+     * Store the new shared post.
+     */
+    public function storeSharePost(ShareRequest $request): Post
+    {
+        $originalPost = Post::find((int)$request->input('original_post_id'));
+
```

Comment:
It would be much better if you put your original_post_id request to the function parameter and create another method so that every method that needs some original posts with original_post_id can use it right? It seems redundant for me. Look at the other method, they are getting the same original posts.

Reply:
I have made changes, Sir. Thank you 🙇